### PR TITLE
simple client filtering on list_jobs

### PR DESCRIPTION
**Pull request by adelebai**

### CODE REVIEW

**Review by sanjgupt — APPROVED** [review]

**Comment by vxfield** [comment]
I added @guenp as a reviewer since she is back as co-owner of the Azure Quantum SDK for Python.
Please wait for her review before merging this.

File: azure-quantum/azure/quantum/job.py [hunk 1]
```diff
@@ -100,6 +102,26 @@ def get_results(self):
...
 
+    def matches_filter(
+        self, 
+        name_match: str = None, 
+        status:  Optional[JobStatus] = None,
+        created_after: Optional[datetime] = None
+    ) -> bool:
+        if name_match is not None:
+            if re.search(name_match, self.details.name) is None:
+               return False
```
Comment (guenp) [inline azure-quantum/azure/quantum/job.py:113]:
```suggestion
        if name_match is not None and re.search(name_match, self.details.name) is None:
           return False
```

File: azure-quantum/azure/quantum/job.py [hunk 2]
```diff
@@ -100,6 +102,26 @@ def get_results(self):
...
+        
+        if status is not None:
+            if self.details.status != status.value:
+                return False
```
Comment (guenp) [inline azure-quantum/azure/quantum/job.py:117]:
```suggestion
        if status is not None and self.details.status != status.value:
            return False
```

File: azure-quantum/azure/quantum/job.py [hunk 3]
```diff
@@ -100,6 +102,26 @@ def get_results(self):
...
+        
+        if created_after is not None:
+            if self.details.creation_time < created_after:
+                return False
```
Comment (guenp) [inline azure-quantum/azure/quantum/job.py:121]:
```suggestion
        if created_after is not None and self.details.creation_time < created_after:
            return False
```

File: azure-quantum/azure/quantum/workspace.py [hunk 4]
```diff
@@ -223,13 +224,28 @@ def get_job(self, job_id: str) -> Job:
...
+        status: Optional[JobStatus] = None,
+        created_after: Optional[datetime] = None
+    ) -> List[Job]:
+        """Returns list of jobs that meet optional (limited) filter criteria. 
+            :param name_match: regex expression for job name matching
+            :param status: filter by job status
+            :param created_after: filter jobs after time of job creation
+
+            todo #27666 - replace some of these with server-side filtering.  
+            todo #27664 - tag filtering when swagger is ready
```
Comment (guenp) [inline azure-quantum/azure/quantum/workspace.py:239]: I would recommend not putting todo's in the code but instead track them in ADO and link this file there

**Review by guenp — CHANGES_REQUESTED** [review]

**Review by guenp — APPROVED** [review]

**Pull request merged**